### ISSUE TITLE
Focus cancellation? Integration with CKEditor as an inline editor

### ISSUE THREAD

**Issue by asafge** [ducksboard/gridster.js#290]
Hey
This might me a symptom of a more general issue, so I thought it would be a good idea to share it with you guys.

CKEditor is a web based text editor, that has a feature called inline editing (demo - http://ckeditor.com/demo#inline). This feature allows a user to click on a html element and turn it into a CKEditor text box. 

I have a client that needs this functionality within gridster, as in "click on a box on the grid and start editing". Unfortunately this is not working seamlessly like it should.

On the gridstar html part, adding 'contenteditable="true" ' to the <li> tags should allow me to start the inline editor on any "box" in the grid, and it does as long as gridster isn't initialized. As soon as I call the gridster() function it will fail to register itself as an inline editor.

After reading their code, it looks like a CKEditor html element needs to be in focus in order to start the inline editing. I had some suspicion that this was an issue with the click event, and as it turns out, by explicitly registering this event and calling focus() on the element, the problem is resolved. Is it possible the gridster suppresses the default focus functionality when clicking on an element?

Note that this is just a left-click issue. When right-clicking the element the inline editor starts as it should. I think there's some sort of collision between the two libraries, and it might happen with other libraries I'm not aware of.

Thanks, and if you need anything else let me know.

**Comment by dartox** [comment]
Hey.

I also use gridster.js but another inline-editor and I have the same problem as you. I followed your instructions and call a focus() on my inline-editing element manually when I click into the gridster-widget.

This strategy works and I'm able to edit the text now, BUT, I am not able to select the text in my element. I can't select it with the mouse or keyboard. Maybe you know why?

Thanks in advance.
Best regards, dartox.

**Comment by asafge** [comment]
Hard to say, can you create a JSFiddle?

**Comment by dartox** [comment]
yes i made a JSFiddle, here it is: http://jsfiddle.net/UVP4N/3/ .... thank you for your help

**Comment by dartox** [comment]
Found a solution. I also have to turn off the "resize" functionality (not just the draggable function) when I click into the editable DIV. Than it works.

**Comment by cyberfuhrer** [comment]
Hi, how to fix this problem? Text in the gridster container not selection

**Comment by dustinbolton** [comment]
Turning off resizing (resize enabled: false) fixes it so editors will work within.  However, it would be very nice to be able to use the resize feature when doing this. This appears to be bug.

**Comment by dustinbolton** [comment]
Note that this bug is not just limited to ckeditor. It also impacts Redactor and likely most/all other editors that require clicking and/or dragging in (eg for highlighting).

**Comment by vieron** [comment]
Related to #334 and #326

**Comment by kunal3** [comment]
Out of curiosity, did someone ever fix this bug or have a fully functional version? If yes, can you link a JSFiddle?
